File: InvolvedEmpireGame/BattleReport.cs
namespace InvolvedEmpireGame
{
    public class BattleReport
    {
        public DateTime DateTime { get; set; }
        public bool AttackSuccessful { get; set; }
        public string Winner { get; set; }
        public string Loser { get; set; }
        public ArmyReport? Attacker { get; set; }
        public ArmyReport? Defender { get; set; }
        public int GoldStolen { get; set; }
        public int MinersStolen { get; set; }
        public int HousesBurnedDown { get; set; }
        
        public BattleReport()
        {
            DateTime = DateTime.Now;
        }
    }
}

File: InvolvedEmpireWebClient/src/app/types/battle-report.ts
import { BattleArmyReport } from "./battle-army-report";

export interface BattleReport {
    dateTime: Date;
    attackSuccessful: boolean;
    winner: string;
    loser: string;
    type: string;

    attacker: BattleArmyReport;
    defender: BattleArmyReport;

    goldStolen: number;
    minersStolen: number;
}